src/subscription_manager/managercli.py

```diff
@@ -2298,6 +2300,28 @@ def _do_command(self):
         if self.options.consumed:
             self.print_consumed(service_level=self.options.service_level, filter_string=self.options.filter_string, pid_only=self.options.pid_only)
 
+    def translate_filter_string(self, filter_string=None):
+        if not filter_string:
+            return None
+        return fnmatch.translate(filter_string)
+
+    def highlight_by_filter_string(self, body, filter_string=None):
+        if not filter_string or not body:
+            return body
+        try:
+            p = re.compile(filter_string, re.IGNORECASE)
+        except Exception, e:
+            log.error("Cannot compile search regex '%s'. %s" % (filter_string, e))
+            return body
+
+        if isinstance(body, list):
```

Comment:
I'd get rid of trying to handle a string or a list of strings if you could (or split it up). Calling code should repr/str/etc before calling it (though that implies those objects have useful repr/str which they likely dont).